build-logic/jvm/src/main/kotlin/gradlebuild.unittest-and-compile.gradle.kts

```
@@ -214,6 +214,12 @@ fun Test.isUnitTest() = listOf("test", "writePerformanceScenarioDefinitions", "w
 
 fun Test.usesEmbeddedExecuter() = name.startsWith("embedded")
 
+fun Test.configureRerun() {
+    outputs.upToDateWhen {
+        !providers.gradleProperty("rerunAllTests").forUseAtConfigurationTime().isPresent
+    }
+}
```

```suggestion
fun Test.configureRerun() {
    if (providers.gradleProperty("rerunAllTests").isPresent) {
        doNotTrackState("All tests should re-run")
    }
}
```

Please use the new `doNotTrackState()` instead of `upToDateWhen { }`. Maybe we need to try this out with test selection, since I am not sure what test selection will do if we don't fingerprint the inputs. I suppose this should be fine, though just in case.

I also think that `forUseAtConfigurationTime()` has been deprecated.